### PR TITLE
Phone Validation on the contact us form.

### PR DESCRIPTION
**Pull request by jpr-odoo**
Task: https://www.odoo.com/web?#id=30897&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
Pad: https://pad.odoo.com/p/r.0ef60fba3a837ccfe5814f24c68c219b

### CODE REVIEW

File: addons/crm_phone_valid/models/crm_lead.py [hunk 1]
```diff
@@ -14,53 +14,47 @@
...
             company = self.env.user.company_id
             try:
-                country = self.country_id and self.country_id.code or company.country_id.code or False
-                nbr = phonenumbers.parse(phone, country)
-            except Exception, e:
-                raise UserError(_('Unable to parse the phone number!'))
+                country = self.country_id and self.country_id.code or company.country_id.code or country
+                nbr = phonenumbers.parse(contact_number, country)
+            except Exception:
+                raise UserError(_('Unable to parse the contact (phone,mobile,fax) number!'))
```
Comment (rim-odoo) [inline addons/crm_phone_valid/models/crm_lead.py:25]: `parse` is rather a technical word... "validate" would be more easy to understand

File: addons/crm_phone_valid/models/crm_lead.py [hunk 2]
```diff
@@ -14,53 +14,47 @@
...
             if not phonenumbers.is_possible_number(nbr):
-                raise UserError(_('Invalid phone number, too few or too much digits!'))
+                raise UserError(_('Invalid contact (phone,mobile,fax) number, too few or too much digits!'))
```
Comment (rim-odoo) [inline addons/crm_phone_valid/models/crm_lead.py:27]:
```python
replace("too much","too many")
```

File: addons/crm_phone_valid/models/crm_lead.py [hunk 3]
```diff
@@ -14,53 +14,47 @@
...
             if not phonenumbers.is_valid_number(nbr):
-                raise UserError(_('Invalid phone number, the prefix does not belong to a valid NPA!'))
+                raise UserError(_('Invalid contact (phone,mobile,fax) number, the prefix does not belong to a valid NPA!'))
```
Comment (rim-odoo) [inline addons/crm_phone_valid/models/crm_lead.py:29]: Ẁhat is a NPA? Sounds like a technical word.

File: addons/website_crm_phone_validate/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        if not model_record:
+            return json.dumps(False)
+
+        try:
+            data = self.extract_data(model_record, request.params)
+            record = data.get('record', {})
+            contact_number = [record.get('phone'), record.get('mobile'), record.get('fax')]
+
+            country = self.get_country()
+            for cantact in contact_number:
```
Comment (rim-odoo) [inline addons/website_crm_phone_validate/controllers/main.py:33]: cantact? probably a typo ;-)

**Review by rim-odoo — CHANGES_REQUESTED** [review]

**Comment by tde-banana-odoo** [comment]
Included in https://github.com/odoo/odoo/pull/16412/files . Thanks !

**Pull request closed without merging**